canonical/websites-content-system
# Create cronjobs

## Done

 - Added cronjobs to:
   - update webpages whose status is `new` to `available` if they exist in the tree structure
   - delete webpages whose status is `to_delete` if they do not exist in the tree structure

## QA

### QA steps

 - Clone the repo and run it using
```bash
docker compose up -d
```
- Open the home page at http://0.0.0.0:8104/ and _wait for the trees to load_
- Check that the docker logs include scheduled tasks
```
docker logs -f websites-content-system-web-1
```
- log into postgres
```bash
docker exec -it websites-content-system-postgres-1 bash
psql -U postgres
```
- Check that there are `available` webpages, considering that all pages were initially `new`.
```
SELECT name, status FROM Webpage WHERE status='AVAILABLE';
```

## Fixes

 - Fixes [WD-13398](https://warthogs.atlassian.net/browse/WD-13398?atlOrigin=eyJpIjoiMzc1NDRkZWQzNjk1NDM2ZjgyM2FmNDBjYTk3NjdkNjEiLCJwIjoiaiJ9)

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.


[WD-13398]: https://warthogs.atlassian.net/browse/WD-13398?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
